### PR TITLE
Feature/bpr 1675

### PR DESCRIPTION
**Pull request by leQu**
I've added a feature that stops the user from sending incomplete data from key-in. When trying to send invoice items with a vat% but without a valid vatID or taxID a modal is shown.

### CODE REVIEW

File: src/client/containers/SalesInvoiceEditor.js [hunk 1]
```diff
@@ -1327,14 +1346,25 @@ class SalesInvoiceEditor extends Components.ContextComponent
...
+            /*
+                [BRP-1675]
+                If used don't have vat or tax ID and tried to add items with a tax amount then save and trigger NoVatpopup
+            */
+             if(button === 'send' 
+                && !this.state.supplier.vatIdentificationNo // supplier[0] for test as admin
+                && !this.state.supplier.taxIdentificationNo // supplier[0] for test as admin
+                && this.state.invoiceTotals.totalTaxAmount>0){
+                this.saveInvoice(invoice, false);
+                new Promise(resolve => setTimeout(resolve, 700)).then(_ => this.showPopupNoVatID())
```
Comment (nczeranka) [inline src/client/containers/SalesInvoiceEditor.js]: This is no reliable approach to deal with concurrency.
If you have to deal with the popup, then you need a different approach.

My proposal to deal with this requirement: Disable the VAT% column and add a hint in the header stating that adding a VAT rate is only possible if the supplier owns a TAX or VAT identifier.

File: src/client/containers/SalesInvoiceEditor.js [hunk 2]
```diff
@@ -1823,7 +1853,6 @@ class SalesInvoiceEditor extends Components.ContextComponent
                             </button>
                         }
                         <button className="btn btn-primary"
-                            disabled={!this.ready4Keyin() || this.state.editMode === "view"}
```
Comment (nczeranka) [inline src/client/containers/SalesInvoiceEditor.js]: Why did you remove it? Does not make sense for me.

Comment (leQu) [reply src/client/containers/SalesInvoiceEditor.js]: You are right, this is not supposed to be removed, was just using it for testing. I'll add it back!

File: package.json [hunk 3]
```diff
@@ -22,20 +22,20 @@
...
   },
   "dependencies": {
+    "@opuscapita/blob-client": "^2.0.11",
+    "@opuscapita/config": "^3.0.21",
     "@opuscapita/db-init": "^2.0.19",
     "@opuscapita/electronic-address": "0.0.2",
     "@opuscapita/event-client": "^2.0.27",
     "@opuscapita/react-select": "^2.0.13",
-    "@opuscapita/service-base-ui": "^1.1.126",
+    "@opuscapita/service-base-ui": "1.1.128",
```
[the remark this answers is not in the document]
Comment (leQu) [reply package.json]: I couldn't get the latest version to work, I'll try again.

File: src/client/containers/SalesInvoiceEditor.js (hunk 1 above)
Comment (leQu) [reply src/client/containers/SalesInvoiceEditor.js]: Thanks for finding this, it was a temp fix that I was going to replace. The idea is to let the user add all the data they want ( including vat%) but when they try to send they will be informed that they are missing a vatID, and since this is done in a modal I didn't find any other standard base UI components that fit and went for another Modal. Any suggestions on how to approach this? My previous solution was to disable the field or to not let the user set the field above 0% without a vatID. Side note: no functionality is connected to the second modal just redirect/close, the save is handled by the other modal.

Comment (leQu) [reply src/client/containers/SalesInvoiceEditor.js]: This is now solved, using a popup message instead of another modal.
